app/ocr.py

```diff
@@ -615,82 +556,100 @@ def get_outcome(self) -> List[str]:
+            #searches the sentence containing file for a state abbrev and returns the first one listed
+            state_matches = similar(file_sentence, state_patterns)
+            if state_matches:
+                return state_matches[0].text
+
+            #if file
+            else:
+                #if you change this output you need to change get_city to reflect that change.
+                return "Please select state"
+        
+        else:
+            #Searches entire document for states using matcher, returns them as a list that we can index. 
+            state_matches = similar(doc, state_patterns)
+            if state_matches[0].text in court_locs.keys():
+                return state_matches[0]
+            else:
+                return "No Matches"
+            
 
     def get_city(self):
-        return "Neu Boston"
-    #     #uses the abbreviation from self.state and #gets the list of cities in
-    #     # a state
-    #     citycache = set()
-    #     if self.state:
-    #         for v in court_locs.get(self.state)['city']:
-    #             citycache.add(v)
-    #
-    #     else:
-    #         for k, v in cour_loc.items():
-    #             for v in court_locs.get(k):
-    #                  citycache.add(v)
-    #
-    #     pattern = [
-    #         [{"LOWER": 'file'}],
-    #         [{"LOWER": 'files'}]
-    #         ]
-    #
-    #     matches = similar(self.doc, pattern)
-    #
-    #     sentence = str(matches[0].sent)
-    #     clean_sentence = sentence.replace(',', ' ').replace('\n', ' ').title()
-    #
-    #     #if an appellate case the first page will have the location of the
-    #     # appellate court, NOT origin
-    #     if self.appellate:
-    #
-    #         #finds first city that's in the citycache
-    #         for word in clean_sentence:
-    #             if word in citycache:
-    #                 return word
-    #
-    #     else:
-    #         #if non-appellate the first page will have the court location
-    #         # including city & state
-    #         for s in self.doc:
-    #             if s in citycache:
-    #                 return s
+        """
+        This function uses the get_state to filter the potential cities to find, then looks for the keywords file or files. 
+            In most cases the city comes right after the keyword. 
+        If the keyword isn't there it then creates a new matcher search for the entire corpus that finds the first city. 
+            This needs to be revised since it will return Falls Church more than it should
+         
+        """
+        #uses the abbreviation from self.state and #gets the list of cities in a state
+        citycache = []
+        city_pattern = []
+        # if get_state() function was unable to find a state it will return "Please select state". If you change this return in get_state you have to change this phrase here to correspond.
+        if self.state == "Please select state":

```

Comment:
I'm guessing this should also check for "No matches", but again, since we're at the end, I'll approve it anyway.

Reply:
I think this would remove the need for the else condition on lines 620-623.